src/parameterized_tendencies/microphysics/precipitation.jl

```diff
@@ -163,16 +163,273 @@ end
+    qᵥ(ts) = TD.vapor_specific_humidity(thp, ts)
+    Lf(ts) = TD.latent_heat_fusion(thp, ts)
+    Tₐ(ts) = TD.air_temperature(thp, ts)
+    #TODO
+    #cv_l(thp) = TD.Parameters.ThermodynamicsParameters.cv_l(thp)
+    cv_l(thp) = thp.cp_l
+    α(ts) = cv_l(thp) / Lf(ts) * (Tₐ(ts) - cmp.ps.T_freeze)
+
+    # compute the precipitation terminal velocity
+    @. ᶜwᵣ[colidx] = CM1.terminal_velocity(
+        cmp.pr,
+        cmp.tv.rain,
+        Y.c.ρ[colidx],
+        Y.c.ρq_rai[colidx] / Y.c.ρ[colidx],
+    )
+    @. ᶜwₛ[colidx] = CM1.terminal_velocity(
+        cmp.ps,
+        cmp.tv.snow,
+        Y.c.ρ[colidx],
+        Y.c.ρq_sno[colidx] / Y.c.ρ[colidx],
+    )
 
-    # TODO in the next PR
-    @. ᶜwᵣ[colidx] = FT(0)
-    @. ᶜwₛ[colidx] = FT(0)
+    # zero out the source terms
     @. ᶜSqₜᵖ[colidx] = FT(0)
     @. ᶜSeₜᵖ[colidx] = FT(0)
     @. ᶜSqᵣᵖ[colidx] = FT(0)
     @. ᶜSqₛᵖ[colidx] = FT(0)
+
+    # All the tendencies are individually limited
+    # by the available condensate (q_ / dt).
+
+    # rain autoconversion: q_liq -> q_rain
+    @. ᶜSᵖ[colidx] = min(
+        qₗ(ᶜts[colidx]) / dt,
+        CM1.conv_q_liq_to_q_rai(
+            cmp.pr.acnv1M,
+            qₗ(ᶜts[colidx]),
+            smooth_transition = true,
+        ),
+    )
+    @. ᶜSqₜᵖ[colidx] -= ᶜSᵖ[colidx]
+    @. ᶜSqᵣᵖ[colidx] += ᶜSᵖ[colidx]
+    @. ᶜSeₜᵖ[colidx] -= ᶜSᵖ[colidx] * (Iₗ(ᶜts[colidx]) + ᶜΦ[colidx])
+
+    # snow autoconversion assuming no supersaturation: q_ice -> q_snow
+    @. ᶜSᵖ[colidx] = min(
+        qᵢ(ᶜts[colidx]) / dt,
+        CM1.conv_q_ice_to_q_sno_no_supersat(
+            cmp.ps.acnv1M,
+            qᵢ(ᶜts[colidx]),
+            smooth_transition = true,
+        ),
+    )
+    @. ᶜSqₜᵖ[colidx] -= ᶜSᵖ[colidx]
+    @. ᶜSqₛᵖ[colidx] += ᶜSᵖ[colidx]
+    @. ᶜSeₜᵖ[colidx] -= ᶜSᵖ[colidx] * (Iᵢ(ᶜts[colidx]) + ᶜΦ[colidx])
+
+    # accretion: q_liq + q_rain -> q_rain
+    @. ᶜSᵖ[colidx] = min(
+        qₗ(ᶜts[colidx]) / dt,
+        CM1.accretion(
+            cmp.cl,
+            cmp.pr,
+            cmp.tv.rain,
+            cmp.ce,
+            qₗ(ᶜts[colidx]),
+            Y.c.ρq_rai[colidx] / Y.c.ρ[colidx],
+            Y.c.ρ[colidx],
+        ),
+    )
+    @. ᶜSqₜᵖ[colidx] -= ᶜSᵖ[colidx]
+    @. ᶜSqᵣᵖ[colidx] += ᶜSᵖ[colidx]
+    @. ᶜSeₜᵖ[colidx] -= ᶜSᵖ[colidx] * (Iₗ(ᶜts[colidx]) + ᶜΦ[colidx])
+
+    # accretion: q_ice + q_snow -> q_snow
+    @. ᶜSᵖ[colidx] = min(
+        qᵢ(ᶜts[colidx]) / dt,
+        CM1.accretion(
+            cmp.ci,
+            cmp.ps,
+            cmp.tv.snow,
+            cmp.ce,
+            qᵢ(ᶜts[colidx]),
+            Y.c.ρq_sno[colidx] / Y.c.ρ[colidx],
+            Y.c.ρ[colidx],
+        ),
+    )
+    @. ᶜSqₜᵖ[colidx] -= ᶜSᵖ[colidx]
+    @. ᶜSqₛᵖ[colidx] += ᶜSᵖ[colidx]
+    @. ᶜSeₜᵖ[colidx] -= ᶜSᵖ[colidx] * (Iᵢ(ᶜts[colidx]) + ᶜΦ[colidx])
+
+    # accretion: q_liq + q_sno -> q_sno or q_rai
+    # sink of cloud water via accretion cloud water + snow
+    @. ᶜSᵖ[colidx] = min(
```

Comment:
Do we need to make sure the snow sink (FT(-1) * ᶜSᵖ[colidx] * α(ᶜts[colidx]) when T > T_freeze) is smaller than qs / dt? Or is it guaranteed?